core/src/stage.cpp

```diff
@@ -687,8 +687,9 @@ bool Connecting::compatible(const InterfaceState& from_state, const InterfaceSta
 
 		for (auto from_it = from_object->shape_poses_.cbegin(), from_end = from_object->shape_poses_.cend(),
 		     to_it = to_object->shape_poses_.cbegin(); from_it != from_end; ++from_it, ++to_it)
-			if (!from_it->matrix().array().isApprox(to_it->matrix().array()))
```

Comment:
You are right as long as matrices are concerned. However, here I turned the matrices into arrays first, which then uses element-wise operations - also for comparison.
However your notation is maybe more readable anyway. Please add spaces around the minus sign.

How do you motivate your choice of 1e-4? This is rather large, isn't it?
Why don't you choose a value close to machine precision as isApprox does?